jsonrpc/transport/websocket.go

```diff
@@ -243,17 +244,20 @@ func (s *stream) setSubscription(id string, callback func(b []byte)) {
 }
 
 // Subscribe implements the PubSubTransport interface
-func (s *stream) Subscribe(method string, callback func(b []byte)) (func() error, error) {
+func (s *stream) Subscribe(ctx context.Context, method string, callback func(b []byte)) error {
```

Comment:
Updated this to synchronize cancellation with the context and avoid race condition to do with closing the channel written to in the callback